src/main/scala/ai/privado/cache/TaggerCache.scala

```diff
@@ -24,32 +24,46 @@
 package ai.privado.cache
 
 import io.shiftleft.codepropertygraph.generated.nodes.{Member, TypeDecl}
-
+import scala.collection.concurrent.TrieMap
 import scala.collection.mutable
-import java.util.concurrent.ConcurrentHashMap
-import scala.jdk.CollectionConverters.ConcurrentMapHasAsScala
 
 class TaggerCache {
   // Stores typeDeclFullName --> ( sourceRuleId --->  Member Node)
-  val typeDeclMemberCache = new ConcurrentHashMap[String, mutable.HashMap[String, mutable.HashSet[Member]]]().asScala
+  val typeDeclMemberCache = TrieMap[String, TrieMap[String, mutable.Set[Member]]]()
```

Comment:
@dbMundada Can we also have a corresponding concurrent DS for `mutable.Set` ?

Reply:
Yes, I also had same thought but got this:

 **scala.collection.mutable.Set[Member]** is used within the **TrieMap**, and access to it is synchronized. This ensures that the inner sets are thread-safe for concurrent updates within the TrieMap, addressing potential concurrency issues when multiple threads modify the same set concurrently.